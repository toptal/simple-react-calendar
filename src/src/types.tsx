import * as Luxon from 'luxon'

export type CalendarSelectionI = CalendarSelectionRangeI | CalendarSelectionSingleI

export interface CalendarRangeSelectedDate extends CalendarSelectionRangeI {
  inProgress: boolean
}

export interface CalendarSelectionRangeI {
  start: Luxon.DateTime
  end: Luxon.DateTime
}

export type CalendarSelectionSingleI = Luxon.DateTime

export enum CalendarSelectionTypeEnum {
  start = 'start',
  end = 'end',
}

export enum CalendarTypeEnum {
  single = 'single',
  range = 'range',
}

export enum MonthHeaderButtonTypeEnum {
  next = 'next',
  prev = 'previous',
}

export enum NoticeTypeEnum {
  overlap = 'overlapping_with_disabled',
  disabled = 'disabled_day_click',
}
