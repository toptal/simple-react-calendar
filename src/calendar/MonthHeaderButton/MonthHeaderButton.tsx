import classnames from 'classnames'
import React, { FC, SyntheticEvent } from 'react'

import { IMonthHeaderButtonRenderProps } from '../../@types'

type Props = {
  blockClassName: string
  customRender?: IMonthHeaderButtonRenderProps
  enabled?: boolean
  onClick: (event: SyntheticEvent<HTMLElement>) => void
  type: 'prev' | 'next'
}

const MonthHeaderButton: FC<Props> = (props) => {
  const {blockClassName, enabled, type, onClick, customRender} = props

  if (customRender) {
    return customRender(props)
  }

  const isNext = type === 'next'

  return (
    <button
      className={classnames(`${blockClassName}-header_button`, `is-${type}`, {
        'is-disabled': !enabled,
      })}
      data-calendar-value={isNext ? 1 : -1}
      disabled={!enabled}
      onClick={onClick}
      title={isNext ? 'Next Month' : 'Previous Month'}
      type="button"
    >
      {isNext ? '<' : '>'}
    </button>
  )
}

MonthHeaderButton.displayName = 'MonthHeaderButton'

export default MonthHeaderButton
