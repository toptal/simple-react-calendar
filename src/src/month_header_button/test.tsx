import {shallow} from 'enzyme'
import * as React from 'react'

import MonthHeaderButton, {MonthHeaderButtonProps} from '.'
import {MonthHeaderButtonTypeEnum} from '../types'

describe('HeaderButton', () => {
  let wrapper, props

  context('when prop `blockClassName` is passed', () => {
    it('renders <MonthHeaderButton />', () => {
      props = getProps({blockClassName: 'example-class'})
      wrapper = shallow(<MonthHeaderButton {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  context('when prop `isDisabled` is passed', () => {
    it('renders <MonthHeaderButton />', () => {
      props = getProps({isDisabled: true})
      wrapper = shallow(<MonthHeaderButton {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  context('when prop `title` is passed', () => {
    it('renders <MonthHeaderButton />', () => {
      props = getProps({title: 'Test Title'})
      wrapper = shallow(<MonthHeaderButton {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  context('when prop `type` is `next`', () => {
    it('renders <MonthHeaderButton />', () => {
      props = getProps({type: MonthHeaderButtonTypeEnum.next})
      wrapper = shallow(<MonthHeaderButton {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})

const getProps = (overrides: Partial<MonthHeaderButtonProps> = {}): MonthHeaderButtonProps => ({
  blockClassName: 'test-class',
  type: MonthHeaderButtonTypeEnum.prev,
  onClick: jest.fn(),
  arrow: shallow(<p key="1">Example</p>),
  ...overrides,
})
