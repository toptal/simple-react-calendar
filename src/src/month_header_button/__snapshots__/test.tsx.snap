// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton when prop \`blockClassName\` is passed renders <MonthHeaderButton /> 1`] = `
<button
  className="example-class-header_button is-prev"
  onClick={[MockFunction]}
  type="button"
  value="previous"
>
  <p
    key="1"
  >
    Example
  </p>
</button>
`;

exports[`HeaderButton when prop \`isDisabled\` is passed renders <MonthHeaderButton /> 1`] = `
<button
  className="test-class-header_button is-prev is-disabled"
  disabled={true}
  onClick={[MockFunction]}
  type="button"
  value="previous"
>
  <p
    key="1"
  >
    Example
  </p>
</button>
`;

exports[`HeaderButton when prop \`title\` is passed renders <MonthHeaderButton /> 1`] = `
<button
  className="test-class-header_button is-prev"
  onClick={[MockFunction]}
  title="Test Title"
  type="button"
  value="previous"
>
  <p
    key="1"
  >
    Example
  </p>
</button>
`;

exports[`HeaderButton when prop \`type\` is \`next\` renders <MonthHeaderButton /> 1`] = `
<button
  className="test-class-header_button is-next"
  onClick={[MockFunction]}
  type="button"
  value="next"
>
  <p
    key="1"
  >
    Example
  </p>
</button>
`;
